### PR TITLE
Implement newEmpty() on MutablePrimitiveCollection. Closes #440.

### LINKED ISSUE #440: Implement newEmpty() on MutablePrimitiveCollection
Similar to `org.eclipse.collections.api.collection.MutableCollection#newEmpty`

### PR DESCRIPTION
**Pull request by donraab**
Signed-off-by: Donald Raab <Donald.Raab@bnymellon.com>

### CODE REVIEW

File: eclipse-collections-code-generator/src/main/resources/impl/stack/mutable/synchronizedPrimitiveStack.stg [hunk 1]
```diff
@@ -419,7 +419,10 @@ public class Synchronized<name>Stack
     @Override
     public Mutable<name>Stack asUnmodifiable()
     {
-        return new Unmodifiable<name>Stack(this);
+        synchronized (this.lock)
+        {
+            return new Unmodifiable<name>Stack(this);
+        }
     }
```
Comment (itohiro73) [inline eclipse-collections-code-generator/src/main/resources/impl/stack/mutable/synchronizedPrimitiveStack.stg]: This fix in `asUnmodifiable()` and the similar fix in `toImmutable()` look good, but these may be better to be made in a separate commit.

I see similar issues still exist in `synchronizedPrimitiveBag.stg`, `synchronizedPrimitiveList.stg` and  `synchronizedPrimitiveSet.stg` as well, so I think they should be fixed altogether in a different commit (not necessarily in this PR).

Comment (nikhilnanivadekar) [reply eclipse-collections-code-generator/src/main/resources/impl/stack/mutable/synchronizedPrimitiveStack.stg]: I would suggest a different PR as well. Helps in compiling the release notes and shows up in history for tracking

**Comment by donraab** [comment]
Removed the fixes to synchronized primitive stack as suggested.  I will submit an issue.

File: eclipse-collections-code-generator/src/main/resources/impl/stack/mutable/synchronizedPrimitiveStack.stg [hunk 2]
```diff
@@ -419,8 +419,8 @@ public class Synchronized<name>Stack
     @Override
     public Mutable<name>Stack asUnmodifiable()
     {
-        return new Unmodifiable<name>Stack(this);
-    }
+            return new Unmodifiable<name>Stack(this);
+        }
```
Comment (itohiro73) [inline eclipse-collections-code-generator/src/main/resources/impl/stack/mutable/synchronizedPrimitiveStack.stg]: It seems the style is broken here. Pls fix.

**Comment by itohiro73** [comment]
LGTM overall besides the style issue I highlighted. Think we can merge it once it's fixed.

File: eclipse-collections-code-generator/src/main/resources/impl/stack/mutable/synchronizedPrimitiveStack.stg (hunk 2 above)
Comment (donraab) [reply eclipse-collections-code-generator/src/main/resources/impl/stack/mutable/synchronizedPrimitiveStack.stg]: Thx, will fix and commit.
